consensus/src/liveness/leader_reputation.rs

```diff
@@ -240,14 +246,37 @@ impl NewBlockEventAggregation {
         history: &'a [NewBlockEvent],
         epoch_to_candidates: &'a HashMap<u64, Vec<Author>>,
         window_size: usize,
+        wrong_order: bool,
```

Comment:
nit: instead of `wrong_order` can we use `reverse_order` to make it look less scary? :)